SkillChat.Client.ViewModel/ColorGetter.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ReactiveUI;
+using SkillChat.Client.ViewModel.Interfaces;
+using Splat;
+
+namespace SkillChat.Client.ViewModel
+{
+    class ColorGetter
+    {
+        private int colorCount = 0;
+        private string[] colorHashs =
+        {
+            "RoyalBlue", "Teal", "SlateBlue", "SlateGray", "DarkSalmon", "Goldenrod", "IndianRed", 
+            "MediumPurple", "DarkRed", "Violet", "OrangeRed"
+        };
+
+        public object GetColor(string userId)
+        {
+            if (!users.ContainsKey(userId))
+            {
+                var colorGenerator = Locator.Current.GetService<IColorGenerator>();
+                var hash = colorHashs[colorCount];
+                if (colorCount < colorHashs.Length - 1)
+                    colorCount++;
+                else
+                    colorCount = 0;
+                var color = colorGenerator.GetColor(hash);
+                users.Add(userId, color);
+            }
+            return users[userId];
+        }
+
+        private Dictionary<string,object> users = new ();
```

Comment:
Для работы в многопоточке лучше использовать [ConcurrentDictionary<TKey,TValue>](https://docs.microsoft.com/ru-ru/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=netcore-3.1)
```suggestion
        private ConcurrentDictionary<string,object> users = new ();
```